mkt/api/middleware.py

```diff
@@ -262,17 +262,27 @@ def process_response(self, request, response):
             getattr(request, 'CORS_HEADERS',
                     ('X-HTTP-Method-Override', 'Content-Type')))
 
-        if fireplacey or getattr(request, 'CORS', None):
+        error_allowed_methods = []
+        # If settings.DEBUG is True, these skip the 404 and 500 handlers
+        # where we set CORS - this happens down in django.core.handlers.
+        #
+        # By setting them here, we ensure developers get the appropriate
+        # headers when testing locally.
+        if (response.status_code in [404, 500] and settings.DEBUG
```

Comment:
Just wondering, zamboni returns quite a few different error codes, would it make sense to do everything 4xx?